tests/unit/charms/tls_certificates_interface/v2/test_tls_certificates_v2_requires.py

```diff
@@ -1146,6 +1146,106 @@ def test_given_csr_in_unit_relation_data_and_certificate_in_remote_relation_data
         assert secret.get_content(refresh=True)["certificate"] == certificate
         assert secret.get_info().expires == expiry_time - timedelta(hours=168)
 
+    def test_given_certificates_available_when_get_assigned_certificates_then_unit_certificates_returned_only(
```

Comment:
Should we have a case that asserts that only the certificates of this unit are returned in case we have more than one unit?
And maybe a case when an empty list is returned when there are no csrs/certficates?